java/gateway-core/src/main/java/com/avaulta/gateway/rules/ColumnarRules.java

```diff
@@ -35,13 +34,29 @@ public class ColumnarRules implements BulkDataRules {
      *
      * in theory, `\t` should also work ...
      */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY) // this doesn't work
+    @NonNull
     @Builder.Default
     protected Character delimiter = ',';
 
+
+    /**
+     * records will be shuffled randomly within chunks of this size, to avoid the original source
+     * file acting as a lookup table for the output file
+     *
+     * (protip: setting to 1 should yield deterministic output for tests)
+     *
+     * TODO: fix serialization of default value
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY) // this doesn't work
+    @NonNull
+    @Builder.Default
+    protected Integer recordShuffleChunkSize = 500;
```

Comment:
yeah, not sure if this is really needed, but ok

Reply:
i agree the number could have a default that comes from a config value somewhere, instead of being passed in via rules. and yeah, we could pick it dynamically or something based on row size to avoid mem problems